Revamp stake pool course | Adding Cardano Node Installation (#849)
cardano-foundation/developer-portal: docs/stake-pool-guide/node-installation-process.md

---
id: node-installation-process
title: Cardano Node Installation process (Environment Setup)
sidebar_label: Cardano Node Installation process
description: Cardano Node Installation process (Environment Setup)
image: ../img/og-developer-portal.png
---
The Cardano node is the core component that underpins the Cardano network. Ultimately, a blockchain network is just a collection of interconnected nodes, all working together to validate transactions and blocks by means of consensus. The definition of consensus for any given network varies, but for the Cardano network it’s defined by the [Ouroboros](stake-pool-basics#understanding-consensus) protocol.

In this section, we will walk you through the process of downloading, compiling, and installing `cardano-node` and `cardano-cli` into your **Linux-based** operating system.

> 

Reviewer: A link & reference needs to be added here, to the most basic (and well maintained) source file for the installation procedure: https://github.com/input-output-hk/cardano-node/blob/master/doc/getting-started/install.md

As commented above regarding the videos, material can go out of date any time the tutorial stops being monitored.  Eventually that will happen this written material: in fact, very quickly after versions of GHC, Cabal, `cardano-node`, libraries, and their package dependencies change.  So since this material is mostly a re-hash of the official installation procedure, they have to be directed there for help & comparison when necessary.

Author: added at the end

Reviewer: 1. The link added at the end is *not* the official instructions for installing the node.  The link I quoted in the comment above is rigorously and responsibly maintained and is the least likely of any document to _ever_ go out of date.  So unless another editor objects, or you can offer a reason why not, I have to insist that this is the reference link included.
2. This link has to go _at the beginning of the document_, not at the end.  The reason is simple: if and when people get into problems months & years down the road, although they may still find this material useful they will need to know of an alternative source *before* they get into trouble.

I'll move & update this reference & comment in another commit (in a few minutes), so we can consider this issue tentatively resolved unless there are any objections.

* the version control system `git`,
* the `gcc` C-compiler,
* C++ support for `gcc`,
* developer libraries for the arbitrary precision library `gmp`,
* developer libraries for the compression library `zlib`,
* developer libraries for `systemd`,
* developer libraries for `ncurses`,
* `ncurses` compatibility libraries,
* the Haskell build tool `cabal`,
* the GHC Haskell compiler (version `8.10.7` or above).

In Redhat, Fedora, and Centos:
```bash
sudo yum update -y
sudo yum install git gcc gcc-c++ tmux gmp-devel make tar xz wget zlib-devel libtool autoconf jq -y
sudo yum install systemd-devel ncurses-devel ncurses-compat-libs -y
```

For Debian/Ubuntu, use the following instead:
```bash
sudo apt-get update -y && sudo apt-get upgrade -y
sudo apt-get install automake build-essential curl pkg-config libffi-dev libgmp-dev libssl-dev libtinfo-dev libsystemd-dev zlib1g-dev make g++ tmux git jq wget libncursesw5 libtool autoconf -y
```
If you are using a different flavor of Linux, you will need to use the correct package manager for your platform instead of `yum` or `apt-get`, and the names of the packages you need to install might differ.

## Installing GHC and Cabal

> The fastest way to install **GHC** (Glasgow Haskell Compiler) and **Cabal** (Common Architecture for Building Applications and Libraries) is to use [ghcup](https://www.haskell.org/ghcup).

Reviewer: ```suggestion
A popular way to install **GHC** (Glasgow Haskell Compiler) and **Cabal** (Common Architecture for Building Applications and Libraries) is to use [ghcup](https://www.haskell.org/ghcup).
```
`ghcup` cannot be the "fastest" way to install GHC because the other way involves unpacking a tarball & running `make install` ... which `ghcup` with its release management also must do.

cd secp256k1
git checkout ac83be33
./autogen.sh
./configure --enable-module-schnorrsig --enable-experimental
make
make check
sudo make install
```

Now we are ready to download, compile and install `cardano-node` and `cardano-cli`. But first, we have to make sure we are back at the root of our working directory:

```bash
cd $HOME/cardano-src
```

Download the `cardano-node` repository:

```bash
git clone https://github.com/input-output-hk/cardano-node.git
cd cardano-node
git fetch --all --recurse-submodules --tags
```
Switch the repository to the latest tagged commit:

```bash
git checkout $(curl -s https://api.github.com/repos/input-output-hk/cardano-node/releases/latest | jq -r .tag_name)
```

:::important
If upgrading an existing node, please ensure that you have read the [release notes on GitHub](https://github.com/input-output-hk/cardano-node/releases) for any changes.
:::

## Configuring the build options

We explicitly use the `ghc` version that we installed earlier. This avoids defaulting to a system version of `ghc` that might be newer or older than the one you have installed.

```bash
cabal configure --with-compiler=ghc-8.10.7
```

> If you are running non x86/x64 platform (eg. ARM) please install and configure LLVM with:

Reviewer: I hadn't know this build target had become standardised, so I'm seeing it here for the first time myself: well done on including it 😎👍
